### ISSUE TITLE
A11y error: The accessibility scan encountered an error. {}

### ISSUE THREAD

**Issue by HosseinRashno** [storybookjs/storybook#12622]
**Describe the bug**
I've followed [this tutoria](https://github.com/storybookjs/storybook/tree/master/addons/a11y#storybook-addon-a11y)l to add **"storybook-addon-a11y"** to my project. Now it shows **"Accessibility"** tab in the addons tabs. It works well for most of the stories, but it shows an error for some of them. This is the error:
```
The accessibility scan encountered an error.
{}
```
The error object is empty so I have no idea what's the problem here. I've tried to add a console log [in this file](https://github.com/donaldpipowitch/storybook/blob/bacccb9fa70859f70aa5438e860156308ffdc26c/addons/a11y/src/components/A11YPanel.tsx) to see the error in the console, and the result was this:
`TypeError {}`

**To Reproduce**
I've added the a11y addon to my project and the error happened!

**Expected behavior**
It should not throw an error for no reason, if something is going wrong, it should show a clear error.

**Screenshots**
![image](https://user-images.githubusercontent.com/9047556/94686861-b9c36700-032b-11eb-882a-08a4f2b08617.png)

**System**
```
  System:
    OS: macOS 10.15.6
    CPU: (8) x64 Intel(R) Core(TM) i5-8257U CPU @ 1.40GHz
  Binaries:
    Node: 12.18.3 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.6 - /usr/local/bin/npm
  Browsers:
    Chrome: 85.0.4183.121
    Firefox: 79.0
    Safari: 13.1.2
  npmPackages:
    @storybook/addon-a11y: ^6.0.22 => 6.0.22
    @storybook/addon-actions: ^6.0.20 => 6.0.20
    @storybook/addon-essentials: ^6.0.20 => 6.0.20
    @storybook/addon-links: ^6.0.20 => 6.0.20
    @storybook/html: ^6.0.20 => 6.0.20
```

**Comment by HosseinRashno** [comment]
I've found the source of the error but still, I can't understand why this code causes an error.
So I use [lit-element](https://lit-element.polymer-project.org/) to create a web-component. This is the `properties` method of the lit-element class component:
```
  static get properties () {
    return {
      href: { type: String },
      type: { type: String },
      attributes: { type: Object }
    }
  }
```
As soon as I remove or rename `attributes` property, the error goes away. I believe this error is related to "storybook" and not "lit-element". I'll keep this issue open for a while to see if anyone knows what's going on here.

**Comment by shilman** [comment]
I agree, we should probably give a better error message here. Any takers?

**Comment by imshubhamsingh** [comment]
@shilman I would want to give it a try. I've been using storybook for quite some time. I would also be good for me to know how storybook works under the hood and give something back to it.

**Comment by shilman** [comment]
@imshubhamsingh 
Thank you for wanting to make Storybook better! 🙏  Please check out [CONTRIBUTING.md](https://github.com/storybookjs/storybook/blob/next/CONTRIBUTING.md#pull-requests-prs) to get started and feel free to ask questions in [our Discord](https://discord.gg/UUt2PJb).

**Comment by imshubhamsingh** [comment]
@shilman I'll debug this and hopefully share information soon on this. Thanks for info ✌️.

**Comment by imshubhamsingh** [comment]
@HosseinRashno can you share a repo which reproduces this error. I created storybook for lit-element but didn't face such an issue.

**Comment by jsomsanith** [comment]
Hi 👋 
Is this still a bug ? If so, can anyone help me reproduce it ?

**Comment by Stunext** [comment]
With Vite builder:

Didn't work on 6.4.18 with old repository "storybook-builder-vite" (Build error)

It worked with the update to "@storybook/builder-vite" and version 6.4.19

Again it stopped working with version 6.4.20 and higher. No compilation errors; but it shows the aforementioned error:

The accessibility scan encountered an error. {}

@jsomsanith https://github.com/Stunext/a11y-bug

**Comment by jsomsanith** [comment]
Hey @Stunext thank you for your a11y-bug repo.
I don't know vite, but from what Ii see is that webpack and vite don't seem to bundle the axe-core commonJS the same way.
In the a11y runner, we first dynamic-import axe-core, then call axe.run on the root element of the story.
- webpack: import('axe-core') ends up with a module containing `{ default: { run, reset, ...}, run, reset, ...}`
- vite: import('axe-core') ends up with a module containing `{ default: { run, reset, ...}}`

The code in axe-core is something like
```
const axe = {};
// a bunch of code adding functions in axe object

module.exports = axe;
```

With both bundler we have the default in the dynamic import module containing all the functions, but with webpack, we also have access to the functions directly. This makes sense as, in theory, we can statically import default containing all the functions, of destructure it to import the named functions.
Don't know why vite ends up with only the default export.

But I'll provide a fix to be compatible with both, as both have access to the module default.

**Comment by IanVS** [comment]
I believe the difference boils down to how vite vs webpack treat compatibility between commonjs and esm.  Technically, according to the spec, `import()` is similar to `import * from`, in the sense that default exports are available on the `default` property.  Webpack (and rollup, which vite uses in production), also seem to check to see if the module being imported was a cjs file initially, and if so, they plop more properties on the results of the `import()`.  

All that said, I think the most correct thing to do here, would be to use the module `default`.  I needed to do something very similar for storybook react: https://github.com/storybookjs/storybook/pull/17987

**Comment by jsomsanith** [comment]
@IanVS Thanks for the explanation and the PR review :)

**Comment by joshrobertsv2** [comment]
@jsomsanith has this fix been released? if so, which release? I am experiencing this issue after switching over to storybook for vite. Any input greatly appreciated!

**Comment by IanVS** [comment]
@joshrobertsv2 it was first included in v6.5.0-beta.0.  If you're having an issue after that version, feel free to open an issue on the vite-builder repo, ideally with a reproduction, and we can take a look.

**Comment by joshrobertsv2** [comment]
Thanks @IanVS !

**Comment by taitems** [comment]
FYI for anyone else who ends up here, I was receiving this on `Storybook 7.0.0-alpha.40 for svelte-vite`, because a default install of this plugin goes to `6.5.x`. Matching the versions made it go away.

```diff
-"@storybook/addon-a11y": "^6.5.13",
+"@storybook/addon-a11y": "^7.0.0-alpha.40",
"@storybook/addon-essentials": "^7.0.0-alpha.40",
"@storybook/addon-interactions": "^7.0.0-alpha.40",
```

**Comment by IanVS** [comment]
@taitems note that you can also use `npx sb@next add @storybook/addon-a11y` to install an addon, which will use the right version, and will add it to your `.storybook/main.js` for you.

**Comment by weo3dev** [comment]
@IanVS - Is your helpful note still applicable with current and RC, and in future, v7 ?  I would find it immensely helpful if a knowledgeable tidbit like this was surfaced a bit more clearly in the main documentation, as I have ended up _here_ specifically, multiple times. Also, it is quite possible I should either decrease or increase intake of caffeine. :: tossup :: 😵‍💫

**Comment by IanVS** [comment]
@weo3dev what are you referring to?  As far as I know, `npx storybook add` has not been changed for 7.0.  @jonniebigodes I wonder if it should be mentioned on https://storybook.js.org/docs/7.0/react/addons/install-addons.

**Comment by weo3dev** [comment]
Oh - apologies for not being clear - using the specific callout of 'sb@next' is really great, and works! Was just wondering if that is a hook, for lack of a better term in my brain, that we will be able to use moving forward to keep add ons up to date more easily.

**Comment by jonniebigodes** [comment]
@IanVS I'm aware of this, and this is something that I'm going to bring over to the team (Storybook & DX) so that we provide a good experience for it. As most addons snippets mention only the package managers (e.g., `yarn`, `npm`, `pnpm) and we probably should start pushing more into using the `add` command the way to go and potentially avoid similar situations. Thanks for putting this on my radar 🙏

**Comment by weo3dev** [comment]
Thank you so much for the clarification! Y'all rock. Thank you so much.
